src/interfaces/IERC7540.sol

```diff
@@ -23,83 +23,100 @@ interface IERC165 {
+     *  unless throwing
+     */
+    function onERC7540RedeemReceived(address _operator, address _sender, uint256 _requestId, bytes memory _data)
+        external
+        returns (bytes4);
 }
 
 interface IERC7540Deposit {
-    event DepositRequest(address indexed sender, address indexed operator, uint256 assets);
+    event DepositRequest(address indexed sender, address indexed receiver, address indexed owner, uint256 assets);
 
     /**
      * @dev Transfers assets from sender into the Vault and submits a Request for asynchronous deposit/mint.
      *
      * - MUST support ERC-20 approve / transferFrom on asset as a deposit Request flow.
      * - MUST revert if all of assets cannot be requested for deposit/mint.
-     * - sender MUST = msg.sender unless some unspecified explicit approval is given by the caller, approval of ERC7540 vault shares is NOT enough.
+     * - owner MUST be msg.sender unless some unspecified explicit approval is given by the caller,
+     *    approval of ERC-20 shares from owner to sender is NOT enough.
      *
-     * @param assets the amount of deposit assets from sender
+     * @param assets the amount of deposit assets to transfer from owner
      * @param receiver the receiver of the request who will be able to operate the request
-     * @param sender the source of the deposit assets
-     * @param bytes additional data which may be used to approve or call the receiver contract
-     * 
+     * @param owner the source of the deposit assets
+     * @param data additional bytes which may be used to approve or call the receiver contract
+     *
      * NOTE: most implementations will require pre-approval of the Vault with the Vault's underlying asset token.
-     * 
-     * Behavior, if data is nonzero, attempt to call the receiver onERC7540DepositReceived otherwise just send the request to the receiver
+     *
+     * If data is nonzero, attempt to call the receiver onERC7540DepositReceived,
+     * otherwise just send the request to the receiver
      */
-    function requestDeposit(uint256 assets, address receiver, address sender, bytes calldata data) external returns (uint256 rid);
+    function requestDeposit(uint256 assets, address receiver, address owner, bytes calldata data)
+        external
+        returns (uint256 rid);
 
     /**
      * @dev Claims a deposit request and sends associated shares to the receiver
      *
      * @param rid the requestId of the deposit request
      * @param receiver the receiver of the claim who will receive the shares output
-     * 
+     * @param owner the owner of the request
+     *
      * @return shares the amount of shares actually received by the claim
      */
-    function claimDeposit(uint256 rid, address receiver, address owner) external returns (uint256 shares);
+    function claimDepositRequest(uint256 rid, address receiver, address owner) external returns (uint256 shares);
```

Comment:
I'm not 100% sure about this, but I didn't really like that we ended up with `claimRedemption` while we use `Redeem` rather than `Redemption` in the rest of the spec, and this was a simple way to make that more consistent